files/delete

```diff
@@ -5,6 +5,14 @@ SHUTDOWN_TIMEOUT=30
 vm_uuid=$1
 vdi_uuid=$2
 
+if [ "$vdi_uuid" ]; then
+    sr_uuid=$(xe vdi-param-get uuid="$vdi_uuid" param-name=sr-uuid)
+    vm_name_label=$(xe vm-param-get uuid="$vm_uuid" param-name=name-label)
+    pbd_uuid=$(xe sr-param-get uuid="$sr_uuid" param-name=PBDs)
+    sr_dir=$(xe pbd-param-get uuid="$pbd_uuid" param-name=device-config param-key=file-uri)
+    dir_links="$sr_dir/directories"
+fi
```

Comment:
You can remove this code if you store the SR path in a file like for VM and VDI uuids. ;)
Search in `files/create`:
```bash
echo -n "$vm_uuid" > "$vm_uuid_file"
echo -n "$vdi_uuid" > "$vdi_uuid_file"
```

Reply:
I see. I am a bit confused if I should use a file or a parameter to pass the sr path. For example, in `files/create`, the script is called with the following parameters:
```bash
"$workpath/container" "$workpath" "$vm_uuid" "$vdi_uuid" "$wrapper_pid" "$container_pipe" &
```
Then, in the `container` script, you have:
```bash
"$workpath/delete" "$vm_uuid" "$vdi_uuid"
```
And finally, in the `delete` script, there is:
```bash
vm_uuid=$1
vdi_uuid=$2 
```
I am probably missing something but what mechanism should I use? Maybe, it is better to use arguments?

Reply:
The goal is to store the SR path in a file, so you can modify this code in `runx` file:
```python
    if [ -f "$crundir/xcp_vm_uuid" ] && [ -f "$crundir/xcp_vdi_uuid" ]; then
        "$workpath/delete" `cat "$crundir"/xcp_vm_uuid` `cat "$crundir"/xcp_vdi_uuid`
        "$workpath/mount" "$crundir" unmount
    fi
```

You can call the delete script with:
```python
"$workpath/delete" "$vm_uuid" "$vdi_uuid" "$sr_path"
```

So to resume:
  - We can store the path in a file during VDI creation
  - It's probably better to call the delete script with an argument path, we can read it like other params in the main script